tests/integration/security/sds_ingress/single_tls_gateway/single_tls_gateway_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package singletlsgateway
+
+import (
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/ingress"
+	ingressutil "istio.io/istio/tests/integration/security/sds_ingress/util"
+
+	"testing"
+)
+
+var (
+	credName = []string{"bookinfo-credential-1"}
+	host     = "bookinfo1.example.com"
+)
+
+// TestSingleTlsGateway_SecretRotation tests a single TLS ingress gateway with SDS enabled.
+// Verifies behavior in these scenarios.
+// (1) when kubernetes secret is not provisioned, which means private key and server certificate
+// are not available. Verifies that listener is not up and connection creation get rejected.
+// (2) A valid kubernetes secret with key/cert is added later, verifies that SSL connection termination is working properly.
+// (3) After key/cert rotation, client needs to pick new CA cert to complete SSL connection. Old CA
+// cert will cause the SSL connection fail.
+func TestSingleTlsGateway_SecretRotation(t *testing.T) {
+	framework.
+		NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			// TODO(JimmyCYJ): Add support into ingress package to test TLS/mTLS ingress gateway in Minikube
+			//  environment
+			if ctx.Environment().(*kube.Environment).Settings().Minikube {
+				t.Skip("https://github.com/istio/istio/issues/14180")
+			}
+
+			ingressutil.DeployBookinfo(t, ctx, g, ingressutil.SingleTLSGateway)
+
+			// Do not provide private key and server certificate for ingress gateway. Connection creation should fail.
```

Comment:
same thing here.

can we do 

```go
testCase := []struct {
  name string
  // other params
} {
  // do each test case.
}
```

Reply:
Good point, I will change it to table-driven test.